glad/documentation.py

```diff
@@ -0,0 +1,171 @@
+                ),
+            )
+
+        # Parameters parsing
+        # DocsGL puts all the function definitions inside .funcsynopsis.funcdef blocks.
+        #
+        # However, instead of describing each function on a separate file, DocsGL combines
+        # multiple related function definitions, whose parameters may be different, into a single
+        # file. This means that we have to find the correct block of parameters for each definition.
+        funcdefs = [
+            d for d in tree.findall('.//*[@class="funcsynopsis"]/*')
+            if d.find('.//*[@class="funcdef"]')
+        ]
+        for func_def in funcdefs:
+            func_name = func_def.find('.//*[@class="fsfunc"]').text
+            func_params = [raw_text(s) for s in func_def.findall('.//var[@class="pdparam"]')]
+
+            # Params are defined in a separate section, often called 'Parameters for <func_name>'
+            # or just 'Parameters'.
+            params_block = next(
+                (s for s in sections if raw_text(s.find('h2')) == f'Parameters for {func_name}'),
+                None,
+            )
+            if not params_block:
+                for p in list(s for s in sections if raw_text(s.find('h2')) == 'Parameters'):
+                    block_params = [raw_text(n) for n in p.findall('.//dt//code')]
+                    if all(p in block_params for p in func_params):
+                        params_block = p
+                        break
+
+            params = []
+            if params_block is not None:
+                for names, desc in zip(
+                    params_block.findall('.//dl//dt'),
+                    params_block.findall('.//dl//dd'),
+                ):
+                    for name in names.findall('.//code'):
+                        param_name = raw_text(name)
+                        if param_name in func_params:
+                            params.append(CommandDocs.Param(param_name, cls.xml_text(desc)))
+            # We interpret params_block=None as a void parameter list.
+
+            commands_parsed[func_name] = CommandDocs(
+                brief, params, description, notes, None, None,
+            )
+        return commands_parsed
+
+    @staticmethod
+    def format(e, is_tail=False):
+        if is_tail:
+            if e.tag == 'mfenced':
+                # closing mathjax fences
+                return f'{e.attrib["close"]}'
+            if e.tag == 'dt':
+                # closing a definition term
+                return '\n'
+            return e.tail
+
+        if e.tag == 'a':
+            return f'![{e.text}]({e.attrib["href"]})'
+        if e.tag == 'code':
+            return f'`{e.text}`'
+        if e.tag == 'mfenced':
+            return f'{e.attrib["open"]}{e.text}'
+        if e.tag == 'dt':
+            return f'\n{CommandDocs.BREAK}- '
+        if e.tag == 'li':
+            return f'\n{CommandDocs.BREAK}-{e.text}'
+        return e.text
+
+    @staticmethod
+    def xml_text(e):
```

Comment:
Could just be a normal free standing function

Reply:
Do you mean an instance method? If so, I can't since I need this in `docs_from_html_file`, which is a `@classmethod` function